core/src/raw/oio/write/chunked_write.rs

```diff
@@ -49,6 +55,12 @@ impl<W: oio::Write> oio::Write for ChunkedWriter<W> {
             self.buffer.advance(written);
         }
 
+        if !self.exact && bs.len() >= self.chunk_size && self.buffer.is_empty() {
```

Comment:
We can calculate if `bs.len() + self.buffer.len() >= self.chunk_size` and write them at once. This could be helpful for cases like: `write(1KiB)`, `write(16MiB)`, `write(1KiB)`.